packages/core/src/RadioCheckToggleCard/RadioCheckToggleCard.tsx

```diff
@@ -0,0 +1,216 @@
+import { themeGet } from '@styled-system/theme-get'
+import { BoxChecked, BoxEmpty, Check, RadioChecked, RadioEmpty } from 'pcln-icons'
+import React from 'react'
+import styled from 'styled-components'
+import { Box, Flex, Text, Toggle } from '..'
+import { getPaletteColor } from '../utils'
+
+export const cardTypes = ['radio', 'checkbox', 'toggle'] as const
+export const hPositions = ['left', 'right'] as const
+export const vPositions = ['top', 'center'] as const
+
+export type TCardTypes = (typeof cardTypes)[number]
+export type THPositions = (typeof hPositions)[number]
+export type TVPositions = (typeof vPositions)[number]
+
+export interface IRadioCheckToggleCard {
+  children?: React.ReactNode | string
+  cardType?: TCardTypes
+  hPosition?: THPositions
+  vPosition?: TVPositions
+  isHeightOccupied?: boolean
+  title: string
+  isTitleBold?: boolean
+  name: string
+  value: string
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
+  isSelected: boolean
+}
+
+const CombinedInput = styled.input`
+  appearance: none;
+  opacity: 0;
+  position: absolute;
+  z-index: 0;
+
+  &:hover + div {
+    --border-color: ${getPaletteColor('primary.base')};
+    --icon-color: ${getPaletteColor('primary.base')};
+  }
+
+  &:focus-visible + div {
+    --icon-border-color: ${getPaletteColor('text.light')};
+    --icon-border-shadow-color: ${getPaletteColor('background.light')};
+  }
+
+  &:active + div {
+    --border-color: ${getPaletteColor('primary.dark')};
+    --icon-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:checked + div {
+    --background-color: ${getPaletteColor('primary.light')};
+    --border-color: ${getPaletteColor('primary.base')};
+    --icon-color: ${getPaletteColor('primary.base')};
+  }
+
+  &:checked:hover + div {
+    --border-color: ${getPaletteColor('primary.dark')};
+    --icon-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:checked:focus-visible + div {
+    --icon-border-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:disabled + div {
+    color: ${getPaletteColor('border.base')};
+  }
+`
+
+const RCTCardContainer = styled(Flex)`
+  --background-color: ${getPaletteColor('background.lightest')};
```

Comment:
Is there an advantage to using CSS variables here? It's a novel pattern for DS, so I want to be sure there's justification for doing it this way.